app/models/mixins/authentication_mixin.rb

```diff
@@ -268,13 +268,12 @@ def authentication_check(*args)
 
   def authentication_check_no_validation(type, options)
     header  = "type: [#{type.inspect}] for [#{id}] [#{name}]"
-    verify_args = self.kind_of?(Host) ? [type, options] : type
     status, details =
       if self.missing_credentials?(type)
         [:incomplete, "Missing credentials"]
       else
         begin
-          verify_credentials(*verify_args) ? [:valid, ""] : [:invalid, "Unknown reason"]
+          verify_credentials(type, options) ? [:valid, ""] : [:invalid, "Unknown reason"]
```

Comment:
> Do you know if there are any cases where self is not a Host and options should not be passed for some specific reason?

No.
My guess is that this `Host` case was added because being over cautious. There is e.g. [ldap_server.rb#verify_credentials](https://github.com/ManageIQ/manageiq/blob/master/app/models/ldap_server.rb#L24) which takes no args and has even this mixin :wink:

I would assume `verify_credentials` should obey the 2 arity api and if something doesnt, it's wrong.